src/CommonGLPI.php

```diff
@@ -755,25 +755,28 @@ public static function createTabEntry($text, $nb = 0, ?string $form_itemtype = n
      *
      * @return void
      **/
-    public function redirectToList()
+    public function redirectToList(): void
+    {
+        Html::redirect($this->getRedirectToListUrl());
+    }
+
+    public function getRedirectToListUrl(): string
     {
         /** @var array $CFG_GLPI */
         global $CFG_GLPI;
 
+        if (!empty($_GET['withtemplate'])) {
+            return $CFG_GLPI["root_doc"] . "/front/setup.templates.php?add=0&itemtype=" . self::getType();
```

Comment:
`getType()` is not final and was overriden in the past by some classes. Maybe it is still the case in some plugins.

```suggestion
            return $CFG_GLPI["root_doc"] . "/front/setup.templates.php?add=0&itemtype=" . static::getType();
```